frontend/src/atoms.tsx

```diff
@@ -192,63 +192,46 @@ export function LoadData(props: { children?: ReactNode }) {
     }
 
     useEffect(() => {
-        let eventDataQueue: WatchEvent[] | undefined = []
+        const eventQueue: WatchEvent[] = []
 
-        async function processEvents() {
-            if (!eventDataQueue) return
-            const dataToProcess = eventDataQueue
-            for (const kind in setters) {
+        function processEventQueue() {
+            if (eventQueue.length === 0) return
+
+            const kindsMap = eventQueue?.reduce((kindsMap, eventData) => {
+                const kind = eventData.object.kind
+                if (kind) {
```

Comment:
Is kind ever falsy?

Reply:
Also, I guess we don't currently support duplicate kinds in different apiGroups?

Reply:
kind should never be falsy, but sometimes I code defensively.
don't currently support duplicate kinds